pkg/mutation/system.go

```diff
@@ -85,16 +86,18 @@ func (s *System) Mutate(obj *unstructured.Unstructured, ns *corev1.Namespace) er
 			if m.Matches(obj, ns) {
 				err := m.Mutate(obj)
 				if err != nil {
-					return errors.Wrapf(err, "Mutation failed for %s %s", obj.GroupVersionKind().Group, obj.GroupVersionKind().Kind)
+					return false, errors.Wrapf(err, "mutation failed for %s %s", obj.GroupVersionKind().Group, obj.GroupVersionKind().Kind)
 				}
 			}
 		}
 		if cmp.Equal(old, obj) {
-			return nil
+			if cmp.Equal(original, obj) {
```

Comment:
I think we can get away with asking if the # of mutation cycles > 1 here to avoid a second deep comparison.

* If we mutate once and don't continue, it's because the mutation didn't do anything.
* If we mutate more than once it's because there was a diff after the first application.
* If we mutate more than once and there was no diff, we have an oscillating system that is non-convergent.

Reply:
Right!
Modified to check only if we are in first cycle.

> If we mutate more than once and there was no diff, we have an oscillating system that is non-convergent.
Let's add this check in a new PR